evergreen/evergreen.yml

```diff
@@ -0,0 +1,381 @@
+    - command: shell.exec
+      params:
+        working_dir: mongo-aspnetcore-odata
+        script: |
+          ${PREPARE_SHELL}
+          OS=${OS} \
+          MONGODB_URI="${MONGODB_URI}" \
+            evergreen/run-pack.sh
+
+  package-smoke-tests:
+    - command: shell.exec
+      type: test
+      params:
+        working_dir: mongo-aspnetcore-odata
+        script: |
+          ${PREPARE_SHELL}
+          OS=${OS} \
+          MONGODB_URI="${MONGODB_URI}" \
+            evergreen/run-smoketests.sh
+
+  package-push:
+    - command: shell.exec
+      params:
+        working_dir: mongo-aspnetcore-odata
+        script: |
+          ${PREPARE_SHELL}
+          OS=${OS} \
+          MONGODB_URI="${MONGODB_URI}" \
+          NUGET_KEY=${nuget_key} \
+            evergreen/run-push.sh
+
+  cleanup:
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          cd "$MONGO_ORCHESTRATION_HOME"
+          # source the mongo-orchestration virtualenv if it exists
+          if [ -f venv/bin/activate ]; then
+            . venv/bin/activate
+          elif [ -f venv/Scripts/activate ]; then
+            . venv/Scripts/activate
+          fi
+          mongo-orchestration stop
+          cd -
+          rm -rf $DRIVERS_TOOLS || true
+
+  fix-absolute-paths:
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          for filename in $(find ${DRIVERS_TOOLS} -name \*.json); do
+            perl -p -i -e "s|ABSOLUTE_PATH_REPLACEMENT_TOKEN|${DRIVERS_TOOLS}|g" $filename
+          done
+
+  windows-fix:
+    - command: shell.exec
+      params:
+        script: |
+          if [ "Windows_NT" = "$OS" ]; then
+            ${PREPARE_SHELL}
+            for i in $(find ${DRIVERS_TOOLS}/.evergreen ${PROJECT_DIRECTORY}/evergreen -name \*.sh); do
+              cat $i | tr -d '\r' > $i.new
+              mv $i.new $i
+            done
+            # Copy client certificate because symlinks do not work on Windows.
+            cp ${DRIVERS_TOOLS}/.evergreen/x509gen/client.pem ${MONGO_ORCHESTRATION_HOME}/lib/client.pem
+          fi
+
+  make-files-executable:
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          for i in $(find ${DRIVERS_TOOLS}/.evergreen ${PROJECT_DIRECTORY}/evergreen -name \*.sh); do
+            chmod +x $i
+          done
+
+pre:
+  - func: fetch-source
+  - func: prepare-resources
+  - func: windows-fix
+  - func: fix-absolute-paths
+  - func: make-files-executable
+  - func: install-dependencies
+
+post:
+  - func: upload-mo-artifacts
+  - func: upload-test-results
+  - func: cleanup
+
+tasks:
+  - name: test-net50
+    commands:
+      - func: bootstrap-mongo-orchestration
+      - func: set-driver-version
+      - func: run-tests
+
+  - name: release-package
+    depends_on:
+      - variant: .tests-variant
+        name: test-net50
+    commands:
+      - func: package-pack
+      - func: upload-package
```

Comment:
Wondering why do we need to upload the package

Reply:
Discussed offline.